site/app/controllers/ChatroomController.php

```diff
@@ -0,0 +1,187 @@
+        $user = $this->core->getUser();
+        $chatrooms = $repo->findAll();
+        $active_chatrooms = $repo->findAllActiveChatrooms();
+
+        if ($user->accessAdmin()) {
+            return new WebResponse(
+                'Chatroom',
+                'showChatPageInstructor',
+                $chatrooms
+            );
+        }
+        else { // Student view
+            return new WebResponse(
+                'Chatroom',
+                'showChatPageStudent',
+                $active_chatrooms
+            );
+        }
+    }
+
+    /**
+     * @Route("/courses/{_semester}/{_course}/chat/{chatroom_id}", methods={"GET"})
+     * @param string $chatroom_id
+     * @return RedirectResponse|WebResponse
+     */
+     public function showChatroom(string $chatroom_id): WebResponse|RedirectResponse {
+         if (!is_numeric($chatroom_id)) {
+            $this->core->addErrorMessage("Invalid Chatroom ID");
+            return new RedirectResponse($this->core->buildCourseUrl(['chat']));
+        }
+
+        $repo = $this->core->getCourseEntityManager()->getRepository(Chatroom::class);
+        $chatroom = $repo->find($chatroom_id);
+
+        return new WebResponse(
+            'Chatroom',
+            'showChatroom',
+            $chatroom
+        );
+     }
+
+    /**
+     * @Route("/courses/{_semester}/{_course}/chat/newChatroom", name="new_chatroom", methods={"POST"})
+     * @AccessControl(role="INSTRUCTOR")
+     */
+    public function addChatroom(): RedirectResponse {
+        $em = $this->core->getCourseEntityManager();
+
+        $chatroom = new Chatroom();
+        $chatroom->setTitle($_POST['title']);
+        $chatroom->setDescription($_POST['description']);
+        $chatroom->setHostId($this->core->getUser()->getId());
+
+        $em->persist($chatroom);
+        $em->flush();
+
+        $this->core->addSuccessMessage("Chatroom successfully added");
+        return new RedirectResponse($this->core->buildCourseUrl(['chat']));
+    }
+
+    /**
+     * @Route("/courses/{_semester}/{_course}/chat/deleteChatroom", methods={"POST"})
+     * @AccessControl(role="INSTRUCTOR")
+     */
+    public function deleteChatroom(): JsonResponse {
+        $chatroom_id = intval($_POST['chatroom_id'] ?? -1);
+        $em = $this->core->getCourseEntityManager();
+
+        /** @var \app\repositories\chat\ChatroomRepository $repo */
+        $repo = $em->getRepository(Chatroom::class);
+
+        /** @var Chatroom|null */
+        $chatroom = $repo->findByChatroomId($chatroom_id);
+
+        if ($chatroom === null) {
+            return JsonResponse::getFailResponse('Invalid Chatroom ID');
+        }
+
+        foreach ($chatroom->getMessages() as $message) {
+            $em->remove($message);
+        }
+        $em->remove($chatroom);
+        $em->flush();
+
+        return JsonResponse::getSuccessResponse();
+    }
+
+/**
+ * @Route("/courses/{_semester}/{_course}/chat/{chatroom_id}/edit", name="edit_chatroom", methods={"POST"})
```

Comment:
We don't ever use Route names so you could remove those from the docblock. Not necessary to remove them but I'm just pointing out that they don't do anything for us. So I'll leave it up to you.